nomad/server.go

```diff
@@ -809,6 +811,14 @@ func (s *Server) Reload(newConfig *Config) error {
 		s.EnterpriseState.ReloadLicense(newConfig)
 	}
 
+	workerPoolArgs := getSchedulerWorkerPoolArgsFromConfigLocked(newConfig)
```

Comment:
This caught my eye and made me go look for a lock, which we don't need because this is a new config. Maybe a quickie comment here to that effect?